contracts/leverage_trading/src/cancel_take_profit_order.rs

```diff
@@ -0,0 +1,158 @@
+use crate::common::Event;
+use crate::execute_order::INACCURACY_RATE;
+use crate::ref_finance::ext_ref_finance;
+use crate::utils::NO_DEPOSIT;
+use crate::*;
+use near_sdk::env::current_account_id;
+use near_sdk::{ext_contract, is_promise_success, Gas, PromiseResult};
+
+#[ext_contract(ext_self)]
+trait ContractCallbackInterface {
+    fn get_take_profit_liquidity_info_callback(
+        &self,
+        order_id: U128,
+        parent_order: Order,
+        take_profit_info: (PricePoints, Order),
+    );
+    fn remove_liquidity_from_take_profit_callback(&self, order_id: U128);
+}
+
+#[near_bindgen]
+impl Contract {
+    pub fn cancel_take_profit_order(&mut self, order_id: U128) {
+        let take_profit_order = self.take_profit_orders.get(&(order_id.0 as u64));
+        require!(take_profit_order.is_some(), "Take profit order not found.");
+
+        let parent_order = self.get_order_by_id(order_id).unwrap().1;
+        let take_profit_order_pair = take_profit_order.unwrap();
+        let tpo = take_profit_order_pair.1.clone();
+        match tpo.status {
+            OrderStatus::PendingOrderExecute => {
+                self.take_profit_orders.remove(&(order_id.0 as u64));
+
+                Event::CancelTakeProfitOrderEvent { order_id }.emit();
+            }
+            OrderStatus::Pending => {
+                ext_ref_finance::ext(self.ref_finance_account.clone())
+                    .with_unused_gas_weight(2)
+                    .with_attached_deposit(NO_DEPOSIT)
+                    .get_liquidity(take_profit_order_pair.1.lpt_id.clone())
+                    .then(
+                        ext_ref_finance::ext(self.ref_finance_account.clone())
+                            .with_unused_gas_weight(1_u64)
+                            .with_attached_deposit(NO_DEPOSIT)
+                            .get_pool(
+                                self.view_pair(&parent_order.sell_token, &parent_order.buy_token)
+                                    .pool_id,
+                            )
```

Comment:
If I'm not mistaken, it won't work. You need to break it down into separate calls and process their results